### DISCUSSION TITLE
Mathematical meaning of wealth index

### DISCUSSION THREAD

**Discussion by alexander-myltsev** [mbk-dev/okama#9] [Q&A] [answered]
I see you're using [rebalanced_portfolio_wealth_ts](https://github.com/mbk-dev/okama/blob/5221bd6281b1ea47c2bf7fe882ba44b22a96af2b/okama/common/helpers.py#L374-L395) at optimizer to build efficient frontier.

I understand `rebalanced_portfolio_wealth_ts` when `period` is `none`. I also understand your idea when `period` is `year`: you simulate the process when an investor waits for the end of the year, then she reinvests the year income to the same portfolio.

A simple numerical experiment as follows:

```python
ls = ['SPY.US', 'GLD.US']
curr = 'USD'
x = ok.AssetList(ls, ccy=curr, last_date='2020-01')
for w in np.linspace(.05, .95, 10):
    weights = [w, 1. - w]
    wealth_ts1 = Rebalance.rebalanced_portfolio_wealth_ts(weights, x.ror, period='none')
    wealth_ts2 = Rebalance.rebalanced_portfolio_wealth_ts(weights, x.ror)
    r1 = wealth_ts1[-1]
    r2 = wealth_ts2[-1]
    print(r2 > r1, round(r1, 4), round(r2, 4))
```

```
output:
True 3329.3155 3394.1949
True 3369.0217 3544.6711
True 3408.728 3668.9166
True 3448.4342 3766.0091
True 3488.1404 3835.2319
True 3527.8466 3876.0821
True 3567.5528 3888.2769
True 3607.259 3871.7586
True 3646.9652 3826.6973
True 3686.6715 3753.4929
```

shows that a yearly rebalanced portfolio *always* has the greater wealth at the end.

Common sense tells me that having the same amount of money at the start and constant weights for the whole backtesting period the result should be equal. Please, explain where extra income comes from in the case of yearly rebalanced portfolio?

**Comment by chilango74** [comment, marked as answer]
Annually rebalanced portfolios do not always have a superior income. You example with 50/50 gold and stocks portfolio is a classic case with low correlation and similar average returns. There are a lot of cases when rebalancing (nor monthly, nor annually) will not give any advantage. 
An example below is MSFT vs BND.
There is a good explanation when rebalancing portfolio has a bonus in Willam Bernstain study: [The rebalancing bonus: Theory and Practice](http://www.efficientfrontier.com/ef/996/rebal.htm).
![image](https://user-images.githubusercontent.com/29829921/112251233-ef5a6480-8c6b-11eb-8009-dc8fae702dc9.png)

**Reply by alexander-myltsev** [reply]
I asked about yearly-rebalanced wealth on its own without relation to Mean-Variance optimization. I expected math formulas as follows:

![image](https://user-images.githubusercontent.com/574267/112446424-3b3b0580-8d61-11eb-86b7-4b1ddacb35c7.png)

which shows that are not ordered in any way in the general case.

Also, your yearly-rebalance wealth reminds me of the [Successive Constant Rebalanced Portfolios (SCRP) algorithm](https://arxiv.org/abs/1212.2129) which has a good performance without extra optimizations.

**Reply by chilango74** [reply]
The chart above is not "optimization" ... there is nothing to optimize when you have 2 assets only. The Efficient frontier just shows all possible portfolios. In the case of `[MSFT.US, BND.US]` not rebalanced portfolios always outperform rebalanced. If you want a particular case, take a 50/50 portfolio:
![image](https://user-images.githubusercontent.com/29829921/112494019-bb7a5e80-8d93-11eb-9379-4fe5d731b65b.png)

The only math study I know about rebalancing and related portfolio performance is a mentioned article by Willam Bernstain...

**Reply by alexander-myltsev** [reply]
What is on the X-axis and what is on the Y-axis in the figure [here](https://github.com/mbk-dev/okama/discussions/9#discussioncomment-521120)?

**Reply by chilango74** [reply]
X-axis is Risk (standard deviation), Y-axis is Return (CAGR). It's a common setup so I just forgot to indicate it.
